packages/libraries/cli/src/helpers/env.ts

```diff
@@ -0,0 +1,17 @@
+/**
+ * Look up a value in the environment if the given value is undefined or null.
+ */
+export const orEnvar = <$Value>(
+  value: $Value,
+  envarName: string,
+): Exclude<$Value, null | undefined> | string | null => {
+  if (value !== null && value !== undefined) return value as any;
```

Comment:
minor: To avoid using `any` you could have a third argument of a function to deserialize the env var into the `$Value` type.
```
<T>(value: T, envName: string, parser: (envValue: string) => T = (v => v)) => ...
```

Alternatively, it may be worth using something like yup or whatever else package we already use to validate the env vars and/or command arguments so that we have an already parsed and valid env to reference

Reply:
The main goal of this PR was to stop class method use.

There are definitely more robust approaches to configuration management. The types of this function at least make the developer aware that a string is always part of the return due to the env lookup, and whatever value they give also adds to the type set. E.g. if their value is `null | Date` they'll end with `null | Date | string` which hopefully forces them to think, or simply make this abstraction better per your comment :)

There are CLI tools (e.g. `@molt/command`, probably `@effect/cli`) that support automatic environment lookup for arguments too, which would be better then the current ad-hoc approach I imagine.

Reply:
Some people on the team view any work on the CLI that isn't Rust based as a waste of time because we want to get to a slim binary for our CLI which isn't easily possible with JS today. Let's see where those conversations go because a lot depends on that I guess. No one on the team likes Oclif for example, but removing it would require addressing the aforementioned point.

Reply:
> every method has the potential to use this, etc., degrades the encapsulation boundaries/scope/purpose etc. of the class to have methods that do not relate to that

Every method could invoke `orEnvar` as well.

> if their value is null | Date they'll end with null | Date | string

I just wanted to let you know that this hypothetical scenario does not need to be addressed right now.